src/zope/app/onlinehelp/browser/tests.py

```diff
@@ -38,53 +78,101 @@ def test_contexthelp(self):
+                                    'id': u'file'
+                                })
 
         self.assertEqual(response.getStatus(), 302)
 
         response = self.publish('/contents.html', basic='mgr:mgrpw')
 
         self.assertEqual(response.getStatus(), 200)
         body = ' '.join(response.getBody().split())
-        self.assert_(body.find(
-            "javascript:popup('contents.html/++help++/@@contexthelp.html") >= 0)
+        self.assertIn(
+            "javascript:popup('contents.html/++help++/@@contexthelp.html",
+            body)
 
         response = self.publish(
             '/contents.html/++help++/@@contexthelp.html', basic='mgr:mgrpw')
 
         self.assertEqual(response.getStatus(), 200)
         body = ' '.join(response.getBody().split())
-        self.assert_(body.find("This is another help!") >= 0)
+        self.assertIn("This is another help!", body)
 
         response = self.publish('/index.html/++help++/@@contexthelp.html',
                                 basic='mgr:mgrpw')
 
         self.assertEqual(response.getStatus(), 200)
         body = ' '.join(response.getBody().split())
-        self.assert_(body.find("This is a help!") >= 0)
+        self.assertIn("This is a help!", body)
 
         response = self.publish('/file/edit.html/++help++/@@contexthelp.html',
                                 basic='mgr:mgrpw')
 
         self.assertEqual(response.getStatus(), 200)
         body = ' '.join(response.getBody().split())
-        self.assert_(body.find(
-            "Welcome to the Zope 3 Online Help System.") >= 0)
+        self.assertIn(
+            "Welcome to the Zope 3 Online Help System.",
+            body)
 
         path = '/contents.html/++help++'
         response = self.publish(path, basic='mgr:mgrpw')
 
         self.assertEqual(response.getStatus(), 200)
         body = ' '.join(response.getBody().split())
-        self.assert_(body.find("Topics") >= 0)
+        self.assertIn("Topics", body)
 
         self.checkForBrokenLinks(body, path, basic='mgr:mgrpw')
 
+class TestZPT(unittest.TestCase):
+
+    layer = OnlineHelpLayer
+
+    def test_render(self):
+        from zope.app.onlinehelp.browser import ZPTOnlineHelpTopicView
+        from zope.publisher.browser import TestRequest
+        from zope.location.interfaces import LocationError
+        from zope.app.rotterdam import Rotterdam
+        from zope.publisher.skinnable import applySkin
+        class Context(object):
+            path = os.path.join(os.path.dirname(__file__),
+                                'helptopic.pt')
+            title = 'title'
+        request = TestRequest()
+        applySkin(request, Rotterdam)
+        zpt = ZPTOnlineHelpTopicView(Context, request)
+        with self.assertRaises(LocationError) as e:
+            zpt.renderTopic()
+        # XXX: How is this supposed to normally get a topicContent
+        # that can be traversed to?
+        self.assertEqual(e.exception.args[1], 'topicContent')
```

Comment:
> There's a ++help++ (helpNamespace), but it doesn't use location proxies. It just assigns to the context attribute of a shared global singleton OnlineHelp instance

Ick. But maybe easy enough to emulate locally in testing since we have broken the (cyclic) dependency on zope.app.apidoc. I'll take a look.

> Threads? Never heard'o'em.

It's only enabled when `devmode` is, right? I wouldn't be surprised if that also tends to single-thread requests.